edit-post/components/sidebar/index.js

```diff
@@ -1,46 +1,45 @@
 /**
  * WordPress Dependencies
  */
-import { __ } from '@wordpress/i18n';
-import { withFocusReturn } from '@wordpress/components';
+import { createSlotFill, ifCondition, withFocusReturn } from '@wordpress/components';
 import { withSelect } from '@wordpress/data';
 import { compose } from '@wordpress/element';
 
 /**
  * Internal Dependencies
  */
 import './style.scss';
-import PostSettings from './post-settings';
-import BlockInspectorPanel from './block-inspector-panel';
-import Header from './header';
+
+const SidebarFill = createSlotFill( 'editPost.Sidebar' );
 
 /**
- * Renders a sidebar with the relevant panel.
- *
- * @param {string} panel The currently active panel.
+ * Renders a sidebar with its content.
  *
  * @return {Object} The rendered sidebar.
  */
-const Sidebar = ( { activeSidebarName } ) => {
+const Sidebar = ( { children, label } ) => {
 	return (
-		<div
-			className="edit-post-sidebar"
-			role="region"
-			aria-label={ __( 'Editor advanced settings' ) }
-			tabIndex="-1"
-		>
-			<Header />
-			{ activeSidebarName === 'edit-post/block' ?
-				<BlockInspectorPanel /> :
-				<PostSettings />
-			}
-		</div>
+		<SidebarFill>
+			<div
+				className="edit-post-sidebar"
+				role="region"
+				aria-label={ label }
+				tabIndex="-1"
+			>
+				{ children }
+			</div>
+		</SidebarFill>
 	);
 };
 
-export default compose(
+const WrappedSidebar = compose(
 	withSelect( ( select ) => ( {
 		activeSidebarName: select( 'core/edit-post' ).getActiveGeneralSidebarName(),
 	} ) ),
+	ifCondition( ( { activeSidebarName, name } ) => activeSidebarName === name ),
```

Comment:
Actually, it'd trigger less re-renders when `isActive` is returned from `withSelect`.